### DISCUSSION TITLE
Forcing IPv4 for LFS HTTPS connections

### DISCUSSION THREAD

**Discussion by FGUFA** [git-lfs/git-lfs#4466] [Q&A]
We are experiencing some troubles using Git LFS via our company VPN, which seem to be caused by the fact that LFS uses IPv6 whenever available.

(Using Git LFS on Windows and a Bitbucket-hosted repository)

Is there any way to force LFS to use only IPv4? Git features the "--ipv4" command line option, but of course this doesn't apply to LFS connections. We can also play with the SSH config, which can be used to force IPv4 for SSH, but again, this doesn't help with LFS HTTPS connections.

**Comment by bk2204** [comment]
Hey,

There isn't such a command-line option.  The reason is that Git LFS is, in the normal course of events, invoked indirectly, often from `git checkout` or `git push`.  There wouldn't be a way to universally invoke it with the `-4` or `--ipv4` option because in many cases, the human isn't the one invoking the process at all and Git has no way to pass those options down.

It should be the case that modern versions of Go use the Happy Eyeballs technique, which falls back to IPv4 automatically if IPv6 is misconfigured, so if you're using an older version of Git LFS, you may want to upgrade to the latest version (usually updating Git for Windows is sufficient) and see if that helps.

If the problem is something different, like that you allow only connections from a certain IP yet your VPN doesn't support IPv6, you could try configuring your routing to drop IPv6 packets temporarily when you're on the VPN, which could make the Happy Eyeballs technique work in this case.  I know Linux has an easy option to disable IPv6 for a network interface, but I'm unsure about how to do that on Windows.

**Reply by FGUFA** [reply]
It is in fact an VPN / IP whitelisting issue, so we don't want Git LFS to use IPv6 even if IPv6 is configured and working on the local machine in general (I can only assume that the Git "--ipv4" option was introduced to deal with similar scenarios).

The problem can be dealt with using OS configuration / routing, but unfortunately this is somewhat messy on Windows. I guess I was hoping for some magical LFS configuration entry "lfs.forceipv4 true" or something similar (effectively disabling Happy Eyeballs).

Thanks for the clarifications!
